lld/test/COFF/arm64ec-lib.test

```diff
@@ -0,0 +1,41 @@
+REQUIRES: aarch64, x86
+RUN: split-file %s %t.dir && cd %t.dir
+
+RUN: llvm-mc -filetype=obj -triple=arm64ec-windows symref.s -o symref-arm64ec.obj
+RUN: llvm-mc -filetype=obj -triple=arm64ec-windows nsymref.s -o nsymref-arm64ec.obj
+RUN: llvm-mc -filetype=obj -triple=aarch64-windows nsymref.s -o nsymref-aarch64.obj
+RUN: llvm-mc -filetype=obj -triple=arm64ec-windows sym.s -o sym-arm64ec.obj
+RUN: llvm-mc -filetype=obj -triple=x86_64-windows sym.s -o sym-x86_64.obj
+RUN: llvm-mc -filetype=obj -triple=aarch64-windows nsym.s -o sym-aarch64.obj
+RUN: llvm-mc -filetype=obj -triple=arm64ec-windows %S/Inputs/loadconfig-arm64ec.s -o loadconfig-arm64ec.obj
+
+RUN: llvm-lib -machine:arm64ec -out:sym-arm64ec.lib sym-arm64ec.obj sym-aarch64.obj
+RUN: llvm-lib -machine:amd64 -out:sym-x86_64.lib sym-x86_64.obj
+
+RUN: lld-link -machine:arm64ec -dll -noentry -out:test.dll symref-arm64ec.obj sym-arm64ec.lib loadconfig-arm64ec.obj
+RUN: lld-link -machine:arm64ec -dll -noentry -out:test2.dll symref-arm64ec.obj sym-x86_64.lib loadconfig-arm64ec.obj
+RUN: lld-link -machine:arm64x -dll -noentry -out:test3.dll symref-arm64ec.obj nsymref-aarch64.obj sym-arm64ec.lib loadconfig-arm64ec.obj
+
+RUN: not lld-link -machine:arm64ec -dll -noentry -out:test-err.dll nsymref-arm64ec.obj sym-arm64ec.lib loadconfig-arm64ec.obj 2>&1 |\
+RUN:              FileCheck --check-prefix=ERR %s
```

Comment:
I'm not quite sure exactly which kind of failure that this test tries to showcase - isn't this the most trivial failure? We've got a library that contains two definitions of `sym` for different architectures, and we have an undefined reference to `nsym`?

Other than that, it'd be nice with a bit more comments about which scenarios the cases above actually tests (what the expected behaviour is, and what the various nuances are). Not sure if it needs to be as comments in the test file, but at least a brief narration of the testcase (as an inline comment or so) would be nice.

Reply:
Regular map in that test has `nsym` symbol, while EC map has `sym` symbol. The test is meant to make sure that we don't resolve the symbol using wrong symbol map. Prior to this PR (or if I messed up ARM64X condition to return early), we'd try to use the regular map in this case and fail with an error that pulled object file is not compatible. It's not a very interesting test, but it seems better than nothing...

I added more comments in tests, thanks.

Reply:
Oh, now I see the source of my confusion. I missed this details: `nsym.s -o sym-aarch64.obj`

To me, I think it'd be clearer if you'd name that object file `nsym-aarch64.obj`, to make it clearer that it defines the symbol `nsym`, not `sym`.